pymodm/base/fields.py

```diff
@@ -61,15 +62,21 @@ def __init__(self, verbose_name=None, mongo_name=None, primary_key=False,
         self.default = default
         # "attname" is the attribute name of this field on the Model.
         # We may be assigned a different name by the Model's metaclass later on.
-        self.attname = mongo_name
-        if self.primary_key and self.mongo_name not in (None, '_id'):
-            raise ValueError(
-                'The mongo_name of a primary key must be "_id".')
-        elif self.primary_key:
-            self.mongo_name = '_id'
+        self.attname = self.mongo_name
         self.__counter = MongoBaseField.__creation_counter
         MongoBaseField.__creation_counter += 1
 
+    def _validate_mongo_name(self, mongo_name):
+        if not self.primary_key and mongo_name == '_id':
+            raise ValueError(
+                '"_id" is reserved as the mongo_name of the primary key.')
```

Comment:
I think this would be improved if it used the option name `primary_key` in the error message, so it's obvious how to fix this without referring to the docs, e.g.

    mongo_name of field <field> is "_id", but primary_key is False

I can see how someone might name a field `_id` because they think they have to define it, and then wind up with this error.